src/app/services/adapter-install.service.ts

```diff
@@ -95,7 +95,14 @@ export class AdapterInstallService {
     });
 
     const exportData = {...adapterConfig, workflow, database, forms, attachments};
-    this.downloadData(exportData, adapterName);
+    return exportData;
+  }
+
+  async exportAdapter(adapterConfig) {
+    const {adapterName} = adapterConfig;
+    this.compileAdapter(adapterConfig).then(exportData => {
+      this.downloadData(exportData, adapterName);
+    });
```

Comment:
I have split the exportAdapter function in two, with the meat moved to "compileAdapter".